### PR TITLE
feat: add organizations as readonly access

### LINKED ISSUE #82: Allow the whole AWS Org for readonly
## Describe the Feature

In scenarios where there are many AWS accounts in the AWS Org, it is often more practical to allow the whole org
to pull images than dozens individual accounts.

## Expected Behavior

From each AWS account users and services can execute non-modifying actions, e.g. pulling the images, list images, etc.

## Describe Ideal Solution

A pseudo-policy:

```
      {
        "Sid": "AllowOrgPull",
        "Effect": "Allow",
        "Principal": "*",
        "Condition": {
          "StringEquals": {
            "aws:PrincipalOrgID": ${jsonencode(data.aws_organizations_organization.current.id)}
          }
        },
        "Action": [
          "ecr:BatchGetImage",
          "ecr:BatchCheckLayerAvailability",
          "ecr:CreateRepository",
          "ecr:DescribeImages",
          "ecr:DescribeImageScanFindings",
          "ecr:DescribeRepositories",
          "ecr:GetAuthorizationToken",
          "ecr:GetDownloadUrlForLayer",
          "ecr:GetLifecyclePolicy",
          "ecr:GetLifecyclePolicyPreview",
          "ecr:GetRepositoryPolicy",
          "ecr:ListImages",
          "ecr:ListTagsForResource"
        ]
      },
```

### PR DESCRIPTION
**Pull request by dragosmc**
## what

* Add the ability to have organizations as trustees (read-only) for the ECR repository

## why

* As described in #82 , it's sometimes useful to allow an entire organization to consume images from a centralized repository

## references

* closes #82

### CODE REVIEW

**Comment by Gowiem** [comment]
@dragosmc we're reworking these policies in #105 -- Let's hold off on this type of work until that is merged and then you can take another crack at this PR. The way policies were originally (or additionally) implemented is leading to some funky code that I want to put a stop to.

**Comment by adamantike** [comment]
It would be great to see this merged!
As a follow up after this one, we can add support for access to specific Organizational Units.

**Comment by Gowiem** [comment]
@dragosmc friendly ping here. Mind fixing conflicts + reworking after #105?

**Comment by dragosmc** [comment]
Ready for a review 👍

File: main.tf
Comment (adamantike) [inline main.tf:296]: I don't think this change is right, as it doesn't allow to provide both principals and organizations with read access for the same ECR repository. Maybe revert this change, and put the organizations read logic in the override list?

**Comment by dragosmc** [comment]
Thanks for the review @adamantike - everythng should be fixed now.

File: main.tf
Comment (adamantike) [inline main.tf:255]: Don't forget to change this to `values = var.organizations_full_access`!

**Comment by dragosmc** [comment]
Hi team, this is ready for another round of review. 🎉

**Comment by adamantike** [comment]
@joe-niland, gentle ping. Can this be reviewed?

**Comment by dragosmc** [comment]
> @dragosmc generally, this looks good and seems to follow the pattern we've set out. Thanks for the updates. To move this forward, please run the following commands, commit, and push the changes:
> 
> ```
> make init
> make github/init
> make readme
> ```
> 
> I'll give it another review once that is done -- Thanks!

Trying to un the above I get the following error:
```
* Installing terraform-docs to /Users/dragos/opensource/terraform-aws-ecr/build-harness/vendor
mkdir -p -m a+rX /Users/dragos/opensource/terraform-aws-ecr/build-harness/vendor
[ -n "/tmp" ] && [ -n "terraform-docs" ] && rm -rf "/tmp/terraform-docs"
mkdir -p /tmp/terraform-docs
curl --retry 3 --retry-delay 5 --fail -sSL -o - https://github.com/segmentio/terraform-docs/releases/download/v0.16.0/terraform-docs-v0.16.0-darwin-arm64.tar.gz | tar -zx -C /tmp/terraform-docs
find /tmp/terraform-docs -type f -name terraform-docs | xargs -I {} cp -f {} /Users/dragos/opensource/terraform-aws-ecr/build-harness/vendor/terraform-docs
chmod +x /Users/dragos/opensource/terraform-aws-ecr/build-harness/vendor/terraform-docs
[ -n "/tmp" ] && [ -n "terraform-docs" ] && rm -rf "/tmp/terraform-docs"
make: gomplate: No such file or directory
make: *** [readme/build] Error 1
```

**Comment by Gowiem** [comment]
@dragosmc I believe the gomplate installation is broken via `make`. Can you please install that tool via whatever platform you're on and try again. If you're on Mac, then `brew install gomplate` will do.

**Comment by max-lobur** [comment]
```
brew install make
make --version
GNU Make 4.4.1

make precommit/terraform # Run dockerized version of precommit hook that we use in CI.
```

**Comment by max-lobur** [comment]
Ensuring make version alone might fix the issue too.

**Comment by dragosmc** [comment]
Thanks @max-lobur @Gowiem . Ensure the correct make version works fine. I see the tests for the readme still fail so not sure what the problem is. Could you have a look please and let me know if there's anything else I need to run?

File: docs/terraform.md
Comment (Gowiem) [inline docs/terraform.md:13]: @dragosmc this change here is what is causing your README failures. I think this may be an issue with an older version of `terraform-docs`. Maybe try upgrading that and then seeing if this gets the changes when you run `make readme` again?

Comment (adamantike) [reply docs/terraform.md:13]: I have experienced this issue in the past, where terraform-docs resolves the current provider version instead of maintaining the version constraint. I have always just reverted those changed lines back, if I'm not touching anything related to providers or dependencies.

**Review by Gowiem — APPROVED** [review]

**Review by max-lobur — APPROVED** [review]

**Pull request merged**
